rotkehlchen/db/snapshots.py

```diff
@@ -202,3 +202,9 @@ def _export(
             return False, str(e)
 
         return True, ''
+
+    def delete(self, timestamp: Timestamp) -> None:
+        """Deletes a snapshot of the database at a given timestamp"""
+        cursor = self.db.conn.cursor()
+        cursor.execute('DELETE FROM timed_balances WHERE time=?', (timestamp,))
+        cursor.execute('DELETE FROM timed_location_data WHERE time=?', (timestamp,))
```

Comment:
It should not happen with the frontend, since the frontend also checks that the timestamp exists but you should also check here that both `execute` statements worked and deleted an entry. You should do that with `cursor.rowcount` checks.

If they did not return an error and return a 409 from the API. Also remember to change the docs to showcase this error can happen.